clang/lib/CIR/Dialect/Transforms/LifetimeCheck.cpp

```diff
@@ -281,6 +287,10 @@ struct LifetimeCheckPass : public LifetimeCheckBase<LifetimeCheckPass> {
   // Track emitted diagnostics, and do not repeat them.
   llvm::SmallSet<mlir::Location, 8, LocOrdering> emittedDiagnostics;
 
+  // Track which values have already been warned about for moved-from state
+  // to avoid multiple warnings for the same variable.
+  llvm::DenseSet<mlir::Value> warnedMovedFromValues;
```

Comment:
Can this be tracked by a location in `emittedDiagnostics` instead?

Reply:
Addressed in c68af59a4b17. Added emittedDiagnostics.clear() in checkFunc() to ensure diagnostics are not suppressed across function boundaries.